mautrix_facebook/portal.py

```diff
@@ -106,6 +107,8 @@ def __init__(self, fbid: str, fb_receiver: str, fb_type: ThreadType,
         self._send_locks = {}
         self._typing = set()
 
+        self._urlRegex = re.compile('https?://(?:[a-zA-Z0-9\-_.~!*\'();:@&=+$,/?%#\[\]])+')
```

Comment:
I came up with this regex for matching URLs with their path and anchor components. It may be a bit permissive, but this is only used for not sending duplicated ShareAttachments